### PR TITLE
Use safe_wraps for wrapping isin()

### LINKED ISSUE #3388: isin issue with numpy 1.12 on Python 2
I received the following report on `da.isin` through a side channel.   cc @jakirkham 

We're using numpy 1.12, which doesn't have numpy.isin().  dask/dask#3363 leads to:

```
% python -c 'import dask.array'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "dask/array/__init__.py", line 8, in <module>
    from .routines import (take, choose, argwhere, where, coarsen, insert,
  File "dask/array/routines.py", line 790, in <module>
    def isin(element, test_elements, assume_unique=False, invert=False):
  File "/opt/python/python-2.7/lib64/python2.7/functools.py", line 33, in update_wrapper
    setattr(wrapper, attr, getattr(wrapped, attr))
AttributeError: 'NoneType' object has no attribute '__module__'
```
where `wrapped` is `None` and `attr` is `'__module__'`.

I think this only happens with Python 2's functools.

py3: https://github.com/python/cpython/blame/0f914b5b5f6ba186afd7112fc851c97247076f70/Lib/functools.py#L52
py2: https://github.com/python/cpython/blame/77f0a41d72886513f5e9277e5587455745c29ac1/Lib/functools.py#L32

### PR DESCRIPTION
**Pull request by shoyer**
Fixes #3388

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

**Comment by shoyer** [comment]
Oops, I accidentally pushed this as a new branch to `dask` rather than my fork. Interestingly CI passed on my branch but not the pull-request...

File: dask/array/routines.py [hunk 1]
```diff
@@ -786,7 +787,14 @@ def _isin_kernel(element, test_elements, assume_unique=False):
     return values.reshape(element.shape + (1,) * test_elements.ndim)
 
 
-@wraps(getattr(np, 'isin', None))
+def safe_wraps(wrapped, assigned=functools.WRAPPER_ASSIGNMENTS):
+    if all(hasattr(wrapped, attr) for attr in assigned):
+        return wraps(wrapped, assigned=assigned)
+    else:
+        return lambda x: x
```
Comment (jakirkham) [inline dask/array/routines.py]: Any thoughts on moving this to a `utils` or similar module?

**Comment by jakirkham** [comment]
Sorry for the delay (was at a conference last week). Looks fine to me. Asked one question above.

File: dask/array/routines.py (hunk 1 above)
Comment (shoyer) [reply dask/array/routines.py]: done

**Comment by jakirkham** [comment]
Thanks @shoyer. :)

**Pull request merged**